scripts/lib/custom_networking.sh

```diff
@@ -0,0 +1,65 @@
+
+    local AZ1=us-west-2a
+    local AZ2=us-west-2b
+    local AZ3=us-west-2c
+
+    CUST_SNET1=$(aws ec2 create-subnet --cidr-block 100.64.0.0/19 --vpc-id $VPC_ID --availability-zone $AZ1 | jq -r .Subnet.SubnetId)
+    CUST_SNET2=$(aws ec2 create-subnet --cidr-block 100.64.32.0/19 --vpc-id $VPC_ID --availability-zone $AZ2 | jq -r .Subnet.SubnetId)
+    CUST_SNET3=$(aws ec2 create-subnet --cidr-block 100.64.64.0/19 --vpc-id $VPC_ID --availability-zone $AZ3 | jq -r .Subnet.SubnetId)
+
+    echo "Subnets - $CUST_SNET1 $CUST_SNET2 $CUST_SNET3"
+    aws ec2 create-tags --resources $CUST_SNET1 --tags Key=kubernetes.io/cluster/$CLUSTER_NAME,Value=shared
+    aws ec2 create-tags --resources $CUST_SNET2 --tags Key=kubernetes.io/cluster/$CLUSTER_NAME,Value=shared
+    aws ec2 create-tags --resources $CUST_SNET3 --tags Key=kubernetes.io/cluster/$CLUSTER_NAME,Value=shared
+
+    local RTASSOC_ID=$(aws ec2 describe-route-tables --filters Name=vpc-id,Values=$VPC_ID |jq -r '.RouteTables[].RouteTableId')
+
+    echo "RT table $RTASSOC_ID"
+
+    aws ec2 associate-route-table --route-table-id $RTASSOC_ID --subnet-id $CUST_SNET1
+    aws ec2 associate-route-table --route-table-id $RTASSOC_ID --subnet-id $CUST_SNET2
+    aws ec2 associate-route-table --route-table-id $RTASSOC_ID --subnet-id $CUST_SNET3
+
+    #Enable custom networking
+    $KUBECTL_PATH set env daemonset aws-node -n kube-system AWS_VPC_K8S_CNI_CUSTOM_NETWORK_CFG=true
+    sleep 5
+    $KUBECTL_PATH set env daemonset aws-node -n kube-system ENI_CONFIG_LABEL_DEF=failure-domain.beta.kubernetes.io/zone
+    echo "Sleep until all aws-nodes are available"
+    sleep 5
+    while [[ $($KUBECTL_PATH describe ds aws-node -n=kube-system | grep "Available Pods: 0") ]]
+    do
+        sleep 5
+        echo "Waiting for daemonset update"
+    done
+    echo "Updated!"
+    
+   sed -i '' -e "s/AZ/$AZ1/g" ./testdata/AZ1_eniconfig.yaml
+   sed -i '' -e "s/CUST_SNET/$CUST_SNET1/g" ./testdata/AZ1_eniconfig.yaml
+
+   sed -i '' -e "s/AZ/$AZ2/g" ./testdata/AZ2_eniconfig.yaml
+   sed -i '' -e "s/CUST_SNET/$CUST_SNET2/g" ./testdata/AZ2_eniconfig.yaml
+
+   sed -i '' -e "s/AZ/$AZ3/g" ./testdata/AZ3_eniconfig.yaml
+   sed -i '' -e "s/CUST_SNET/$CUST_SNET3/g" ./testdata/AZ3_eniconfig.yaml
+
+  $KUBECTL_PATH apply -f ./testdata/AZ1_eniconfig.yaml
+  $KUBECTL_PATH apply -f ./testdata/AZ2_eniconfig.yaml 
+  $KUBECTL_PATH apply -f ./testdata/AZ3_eniconfig.yaml
+
+  #no need to set max-pods since it is already configured while creating the ASG
+  INSTANCE_IDS=(`aws ec2 describe-instances --query 'Reservations[*].Instances[*].InstanceId' --filters "Name=tag-key,Values=eks:cluster-name" "Name=tag-value,Values=$CLUSTER_NAME" --output text` )
+  for i in "${INSTANCE_IDS[@]}"
+  do
+	    echo "Terminating EC2 instance $i ..."
+	    aws ec2 terminate-instances --instance-ids $i
+  done
+  echo "Terminated instances successfully. Now wait 400s to let ASG create a new instance."
+  sleep 400
```

Comment:
Is there an API to check if it's deleted? Sleep isn't really deterministic

Reply:
Right, I will add an helper to verify nodes are up.